andela/Ah-backend-xmen
# # 162725259: Users should be able to create  articles

## What does this PR do?
This PR is to create the Articles endpoint to enable users creater, delete, read and update articles

### Description of Task to be completed?
Articles form the backbone of our application and introduce us to CRUD which is the building block of most web applications.

### How should this be manually tested?
- Fetch this branch in your terminal by running the command `git fetch origin ft-create-an-article-162725259`
- Check into the branch by running `git checkout ft-create-an-article-162725259`
- Activate your virtual environment by running `source venv/bin/activate`
- Install dependencies by running `pip install -r requirements.txt`
- Migrate by running `python manage.py migrate`
- Run `python manage.py run server`
- Open postman to test the endpoints as shown in the screen shots below

### POST
<img width="1117" alt="screen shot 2019-01-21 at 12 36 06 pm" src="https://user-images.githubusercontent.com/39744587/51465560-2e818800-1d79-11e9-8639-3ff73ddbe1f6.png">

### GET
<img width="1105" alt="screen shot 2019-01-21 at 12 34 40 pm" src="https://user-images.githubusercontent.com/39744587/51465474-0db93280-1d79-11e9-934d-6fea9e4df6f7.png">

### PATCH
<img width="1133" alt="screen shot 2019-01-21 at 10 29 30 am" src="https://user-images.githubusercontent.com/39744587/51458796-7f888080-1d67-11e9-9e45-1fae25aadccb.png">

### DELETE
<img width="1133" alt="screen shot 2019-01-21 at 3 37 32 pm" src="https://user-images.githubusercontent.com/39744587/51475125-91334d80-1d92-11e9-9b66-ee0d5de2bc1c.png">

What are the relevant pivotal tracker stories?
[#162725259](https://www.pivotaltracker.com/n/projects/2232225/stories/162725259)

### POSTMAN
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/7be0f4c7b7a6a9d6c211)
